myfitnesspal/client.py

```diff
@@ -830,23 +827,23 @@ def set_new_food(
         fat: float,
         carbs: float,
         protein: float,
-        sodium: float = "",
-        potassium: float = "",
-        saturated_fat: float = "",
-        polyunsaturated_fat: float = "",
-        fiber: float = "",
-        monounsaturated_fat: float = "",
-        sugar: float = "",
-        trans_fat: float = "",
-        cholesterol: float = "",
-        vitamin_a: float = "",
-        calcium: float = "",
-        vitamin_c: float = "",
-        iron: float = "",
+        sodium: float = 0,
+        potassium: float = 0,
+        saturated_fat: float = 0,
+        polyunsaturated_fat: float = 0,
+        fiber: float = 0,
+        monounsaturated_fat: float = 0,
+        sugar: float = 0,
+        trans_fat: float = 0,
+        cholesterol: float = 0,
+        vitamin_a: float = 0,
+        calcium: float = 0,
+        vitamin_c: float = 0,
+        iron: float = 0,
         serving_size: str = "1 Serving",
         servingspercontainer: float = 1.0,
         sharepublic: bool = False,
-    ) -> bool:
+    ) -> None:
```

Comment:
I looked through the method, and it seemed like there were two possibilities: we returned `True` or we raised an exception.  The `bool` return was really just communicating that the operation was successful, and we already know the operation was successful if an exception wasn't raised.

Reply:
I understand why you replaced the empty strings to 0. But in most scenarios this is just not the same and wrong. If we want to add a food item to MFP and there is no value for example fiber on the package it does not mean that there is no fiber (0) in it, it is just not provided or unknown. MFP was able to handle the empty post for such micro nutritions and it was correctly shown as “-” on MFP. Now there is a false “0” shown. Unfortunately I don't know what is the best solution for this problem.

To change the return value to None looks fine.

Reply:
So, if it's not clear -- the reason that I changed those is that `""` is not a `float`, which was the type you declared.  Typing tools like `mypy` won't allow you to do things like that.  If we do want to send `""` when a value isn't provided, we probably want to use instead `Optional[float] = None` and translate those `None` values into `""` when sending them to MFP.

Reply:
It was clear. However it worked so far, but you are right it was not "clean".
I implemented your suggestion. ;)